internal/common/common.go

```diff
@@ -0,0 +1,97 @@
+package common
```

Comment:
Try not to have a package named `common` or `utils`.

Consider
- `package slice` for `func Contains[E comparable](s []E, v E) bool` (originally `IsPresent`)
- `package pkix` for DN related methods
- `package registry` for `ValidateRegistryScopeFormat`

If internal methods are used within a package, then an unexported method will be better.